### PR TITLE
convert api/media tests to pytest

### PR DESCRIPTION
**Pull request by peterdragun**
Test results:
```
pytest -v tests/foreman/api/test_media.py
========================================== test session starts ==========================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 -- 
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir: 
plugins: forked-1.0.2, xdist-1.29.0, mock-1.10.4, cov-2.7.1, services-1.3.1
collecting ... 2020-08-10 11:51:40 - conftest - DEBUG - Collected 60 test cases
collected 60 items                                                                                      

tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[0] PASSED              [  1%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[1] PASSED              [  3%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[2] PASSED              [  5%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[3] PASSED              [  6%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[4] PASSED              [  8%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[5] PASSED              [ 10%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_name[6] PASSED              [ 11%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[0] PASSED         [ 13%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[1] PASSED         [ 15%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[2] PASSED         [ 16%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[3] PASSED         [ 18%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[4] PASSED         [ 20%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[5] PASSED         [ 21%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[6] PASSED         [ 23%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[7] PASSED         [ 25%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[8] PASSED         [ 26%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[9] PASSED         [ 28%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os_family[10] PASSED        [ 30%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_location PASSED             [ 31%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os PASSED                   [ 33%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[0] PASSED      [ 35%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[1] PASSED      [ 36%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[2] PASSED      [ 38%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[3] PASSED      [ 40%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[4] PASSED      [ 41%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[5] PASSED      [ 43%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[6] PASSED      [ 45%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[7] PASSED      [ 46%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[8] PASSED      [ 48%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[9] PASSED      [ 50%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_url PASSED          [ 51%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_os_family PASSED    [ 53%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[0] PASSED                   [ 55%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[1] PASSED                   [ 56%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[2] PASSED                   [ 58%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[3] PASSED                   [ 60%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[4] PASSED                   [ 61%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[5] PASSED                   [ 63%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_name[6] PASSED                   [ 65%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_url PASSED                       [ 66%]
tests/foreman/api/test_media.py::TestMedia::test_positive_update_os_family PASSED                 [ 68%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[0] PASSED                   [ 70%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[1] PASSED                   [ 71%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[2] PASSED                   [ 73%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[3] PASSED                   [ 75%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[4] PASSED                   [ 76%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[5] PASSED                   [ 78%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[6] PASSED                   [ 80%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[7] PASSED                   [ 81%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[8] PASSED                   [ 83%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[9] PASSED                   [ 85%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_url PASSED                       [ 86%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_os_family PASSED                 [ 88%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[0] PASSED                        [ 90%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[1] PASSED                        [ 91%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[2] PASSED                        [ 93%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[3] PASSED                        [ 95%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[4] PASSED                        [ 96%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[5] PASSED                        [ 98%]
tests/foreman/api/test_media.py::TestMedia::test_positive_delete[6] PASSED                        [100%]
================================ 60 passed, 4 warnings in 137.34 seconds ================================
```

### CODE REVIEW

**Comment by peterdragun** [comment]
Positive tests were squashed as requested. I don't see any benefit from merging negative tests into one, so I kept them as before with just small change - shared media for negative updates.

New test results:

```
pytest -v tests/foreman/api/test_media.py 
=========================================== test session starts ============================================
platform linux -- Python 3.7.8, pytest-4.6.3, py-1.8.0, pluggy-0.12.0 -- 
cachedir: .pytest_cache
shared_function enabled - OFF - scope:  - storage: file
rootdir:
plugins: forked-1.0.2, xdist-1.29.0, mock-1.10.4, cov-2.7.1, services-1.3.1
collecting ... 2020-08-13 10:13:58 - conftest - DEBUG - Collected 45 test cases
collected 45 items                                                                                         

tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[0] PASSED                   [  2%]
tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[1] PASSED                   [  4%]
tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[2] PASSED                   [  6%]
tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[3] PASSED                   [  8%]
tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[4] PASSED                   [ 11%]
tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[5] PASSED                   [ 13%]
tests/foreman/api/test_media.py::TestMedia::test_positive_crud_with_name[6] PASSED                   [ 15%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[0] PASSED     [ 17%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[1] PASSED     [ 20%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[2] PASSED     [ 22%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[3] PASSED     [ 24%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[4] PASSED     [ 26%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[5] PASSED     [ 28%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[6] PASSED     [ 31%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[7] PASSED     [ 33%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[8] PASSED     [ 35%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[9] PASSED     [ 37%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_with_os_family[10] PASSED    [ 40%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_location PASSED                [ 42%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_with_os PASSED                      [ 44%]
tests/foreman/api/test_media.py::TestMedia::test_positive_create_update_url PASSED                   [ 46%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[0] PASSED         [ 48%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[1] PASSED         [ 51%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[2] PASSED         [ 53%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[3] PASSED         [ 55%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[4] PASSED         [ 57%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[5] PASSED         [ 60%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[6] PASSED         [ 62%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[7] PASSED         [ 64%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[8] PASSED         [ 66%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_name[9] PASSED         [ 68%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_url PASSED             [ 71%]
tests/foreman/api/test_media.py::TestMedia::test_negative_create_with_invalid_os_family PASSED       [ 73%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[0] PASSED                      [ 75%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[1] PASSED                      [ 77%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[2] PASSED                      [ 80%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[3] PASSED                      [ 82%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[4] PASSED                      [ 84%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[5] PASSED                      [ 86%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[6] PASSED                      [ 88%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[7] PASSED                      [ 91%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[8] PASSED                      [ 93%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_name[9] PASSED                      [ 95%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_url PASSED                          [ 97%]
tests/foreman/api/test_media.py::TestMedia::test_negative_update_os_family PASSED                    [100%]

================================= 45 passed, 4 warnings in 113.79 seconds ==================================

```

**Comment by tstrych** [comment]
@mirekdlugosz can you please look once again :) Thank you

File: tests/foreman/api/test_media.py [hunk 1]
```diff
@@ -16,60 +16,73 @@
...
         :expectedresults: Media entity is created and has proper OS family
             assigned
         """
-        for os_family in OPERATING_SYSTEMS:
-            with self.subTest(os_family):
-                media = entities.Media(organization=[self.org], os_family=os_family).create()
-                self.assertEqual(media.os_family, os_family)
+        media = entities.Media(organization=[module_org], os_family=os_family).create()
+        assert media.os_family == os_family
+        new_os_family = OPERATING_SYSTEMS[random.randint(1, len(OPERATING_SYSTEMS) - 1)]
```
Comment (mirekdlugosz) [inline tests/foreman/api/test_media.py:80]: `random.choice()` returns random element from iterable:
```suggestion
        new_os_family = random.choice(OPERATING_SYSTEMS)
```

**Review by mirekdlugosz — APPROVED** [review]
ACK with suggestion

**Review by pondrejk — APPROVED** [review]

**Pull request merged**